src/compose/volume-manager.ts

```diff
@@ -23,9 +24,29 @@ export async function get({ name, appId }: VolumeNameOpts): Promise<Volume> {
 	);
 }
 
-export async function getAll(): Promise<Volume[]> {
-	const volumeInspect = await listWithBothLabels();
-	return volumeInspect.map((inspect) => Volume.fromDockerVolume(inspect));
+export async function getAll(allowLegacy = false): Promise<Volume[]> {
+	let volumes;
+	if (allowLegacy) {
+		// Returns all Volumes which might not have a supervised label
+		volumes = await list();
+	} else {
+		// Only get Volumes with supervised label
+		volumes = await listWithBothLabels();
+	}
+	// Normalize inspect information to Volume types and filter any that fail
+	return volumes.reduce((volumesList, volumeInfo) => {
```

Comment:
@Page- what are your thoughts on a cleaner way of doing this. I never check what options lodash has so maybe there is one. To be clear, I'm happy with this solution but I think you've pointed out smaller LOC implementations that achieve the same thing. I basically need a filter catch function..